### PR TITLE
Add attention in circle-ci

### PR DESCRIPTION
**Pull request by ngeiswei**

### CODE REVIEW

**Comment by ngeiswei** [comment]
Circle-CI fails with
```
Error applying workspace layer for job f62cb2b2-420c-4e63-9f60-a6f9bbfa6b30: Concurrent upstream jobs persisted the same file(s)
```
Anyone, any idea?

File: .circleci/config.yml [hunk 1]
```diff
@@ -103,6 +103,113 @@ jobs:
...
+          command: cd build && make -j2
+      - run:
+          name: Build tests
+          command: cd build && make -j2 tests
+      - run:
+          name: Run tests
+          command: cd build && make -j2 test
+      - run:
+          name: Install URE
+          command: cd build && make -j2 install && ldconfig
```
Comment (vsbogd) [inline .circleci/config.yml]: @ngeiswei , I believe this code can be the reason of CircleCI error.

File: .circleci/config.yml [hunk 2]
```diff
@@ -103,6 +103,113 @@ jobs:
...
```
[the remark this answers is not in the document]
Comment (vsbogd) [reply .circleci/config.yml]: Second look shows that `Install URE` section in fact just reinstalls `attention` and `cogserver`, so it should not be an issue.

[the remark this answers is not in the document]
Comment (vsbogd) [reply .circleci/config.yml]: @ngeiswei , in my comments above I mean the following change https://github.com/opencog/cogutil/pull/202/commits/08bf715f516e3feb0b6844a3280bffc6a05910c0

**Comment by vsbogd** [comment]
@ngeiswei , I have found that this commit https://github.com/opencog/cogutil/pull/204/commits/57166dd2e89a4dd573bafea1f95441d0271f3775 solves the issue. It seems that copying `ccache` into workspace in two parallel jobs lead to the conflict when `cache` is applied to the `opencog` job workspace.

**Comment by ngeiswei** [comment]
Thanks @vsbogd , though it's strange that circle-ci is that brittle, maybe we're not doing something properly.

**Comment by ngeiswei** [comment]
OK, it makes sense that circleci would complain about writing files concurrently.

Till we have a better solution, I guess disabling copying ccache files here and there is the right thing to do. Still I'm puzzled, that must be a pretty typical problem, there must be a proper solution to this.

**Comment by ngeiswei** [comment]
Could it be a bug from ccache

File: .circleci/config.yml [hunk 3]
```diff
@@ -22,13 +22,13 @@ jobs:
           command: mkdir build && cd build && cmake ..
       - run:
           name: Build
-          command: cd build && make -j4
+          command: cd build && make -j2
```
Comment (vsbogd) [inline .circleci/config.yml]: BTW, it probably makes sense to use `MAKEFLAGS` environment variable in CircleCI job settings to get rid of adding this everywhere.

**Pull request merged**